### PR TITLE
91 edit comment to post

### LINKED ISSUE #91: edit comment to post

### PR DESCRIPTION
**Pull request by Nour-Agha99**

### CODE REVIEW

**Review by Shatha-Amin — APPROVED** [review]

**Review by mobatra — APPROVED** [review]

File: src/controllers/comments/editCommentController.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    const { content, commentId } = req.body;
+    await editCommentSchema.validateAsync({ content, commentId }, { abortEarly: false });
+
+    const { rows: comment } = await getCommentByIdQuery({ commentId });
+
+    if (!comment.length) {
+      throw new CustomError('NOT FOUND', 404);
+    }
+    if (comment[0].user_id !== req.user?.id) {
+      throw new CustomError('NOT FOUND', 404);
```
Comment (karamzomlot) [inline src/controllers/comments/editCommentController.ts]: Unauthorized error

File: src/controllers/comments/editCommentController.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const editCommentController = async (
+  req: EditCommentRequest,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    if (!req.user) throw new CustomError('Unauthorized', 401);
+
+    const { content, commentId } = req.body;
```
Comment (karamzomlot) [inline src/controllers/comments/editCommentController.ts]: set the commentId in the param

File: src/routes/comments.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import express, { Router } from 'express';
+import { addCommentController, editCommentController, getCommentsController } from '../controllers';
+import checkAuth from '../middlewares';
+
+const commentsRouter: Router = express.Router();
+
+commentsRouter.use(checkAuth);
+commentsRouter.get('/', getCommentsController);
+commentsRouter.post('/', addCommentController);
+commentsRouter.put('/', editCommentController);
```
Comment (karamzomlot) [inline src/routes/comments.ts]: set id param for the comment here

File: src/utils/types/requests/index.ts [hunk 4]
```diff
@@ -51,6 +59,14 @@ interface EditPostRequest extends RequestWithDecoded {
   };
 }
 
+interface EditCommentRequest extends RequestWithDecoded {
+  body: {
+    content: string;
+    postId: number;
+    commentId: number;
+  };
```
Comment (karamzomlot) [inline src/utils/types/requests/index.ts]: don't forget to add the commentId here as a param, not in the req.body

**Review by karamzomlot — CHANGES_REQUESTED** [review]

**Review by KhaledAref13 — APPROVED** [review]

**Review by karamzomlot — APPROVED** [review]

**Pull request merged**